apps/throttling.py

```diff
@@ -0,0 +1,57 @@
+import logging
+from calendar import monthrange
+from datetime import datetime
+from influxdb_metrics.loader import log_metric
+
+from rest_framework import throttling
+
+LOG = logging.getLogger('transmission')
+
+
+# pylint: disable=attribute-defined-outside-init, super-init-not-called
+class MonthlyRateThrottle(throttling.SimpleRateThrottle):
+    def __init__(self):
+        month = datetime.now().month
+        year = datetime.now().year
+        self.duration = monthrange(year, month)[1] * 86400
+        # Rate limit is dynamically extracted from JWT
+
+    def get_cache_key(self, request, view):
+        if not request.user.is_authenticated:
+            return None
+
+        return request.user.token.get('organization_id', None)
+
+    def allow_request(self, request, view):
+        LOG.debug('Checking request for throttling')
+
+        if request.method == 'GET':
+            return True
+
+        self.key = cache_key = self.get_cache_key(request, view)
```

Comment:
Most of these local variables you define like `self.key` don't actually need to be on `self`, since they are local to the function. If you remove `self`, you can probably get rid of the pylint disable for `attribute-defined-outside-init`

Reply:
those are actually needed to be defined there, in order for the throttle_success to successfully add to the cache

Reply:
nvm since I'll be adding overriding the throttle_success/failure

Reply:
Oh actually if they're used there, keep them in self and initialize them in __init__. I was more referring to self.num_requests for example.

Reply:
You marked this as resolved but did not address it (`self.num_requests`)